### ISSUE TITLE
extensible frontmatter schema

### ISSUE THREAD

**Issue by pelikhan** [microsoft/prompty#113]
We would like to store more data in the frontmatter, like "url" or "source". Is it possible to make it extensible or can we submit new optional fields?

**Comment by sethjuarez** [comment]
I see - as part of the metadata? This is a good suggestion.

**Comment by pelikhan** [comment]
Yes, the frontmatter can be useful to store a lot of custom metadata. Currently the schema does not allow additional properties.

**Comment by sethjuarez** [comment]
I moved to a [TypeSpec definition](https://github.com/microsoft/prompty/blob/dev/vscode/prompty/models/prompty.tsp) to make it all much better. The idea is to automatically generate some of the base classes (eventually) in multiple languages so all the loaders work the same. The [latest dev release](https://pypi.org/project/prompty/1.0.0a2/) has the majority of the new schema implemented (finalizing the output properties this week).
